src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -469,6 +463,32 @@ protected function initializeBundles()
     }
 
     /**
+     * @param BundleInterface $bundle
+     * @param bool $dependency
+     */
+    protected function addBundle($bundle, $dependency = false)
+    {
+        $name = $bundle->getName();
+
+        if (isset($this->bundles[$name]) && false === $dependency) {
```

Comment:
Yea but as I said above, the native behavior from Symfony does not allow using two bundles with the same name, even if they have different FQCN. It is useless to implement this feature in this PR, because it would need to refactor the bundle registration system in order to "change" a bundle name if another one exists with the same name but with another FQCN. And this is not the point of my implementation.